src/native/eventpipe/ds-dump-protocol.h

```diff
@@ -61,5 +61,27 @@ ds_dump_protocol_helper_handle_ipc_message (
 	DiagnosticsIpcMessage *message,
 	DiagnosticsIpcStream *stream);
 
+/*
+* DiagnosticsGenerateCoreDumpResponsePayload
+*/
+
+#if defined(DS_INLINE_GETTER_SETTER) || defined(DS_IMPL_DUMP_PROTOCOL_GETTER_SETTER)
+struct _DiagnosticsGenerateCoreDumpResponsePayload {
+#else
+struct _DiagnosticsGenerateCoreDumpResponsePayload_Internal {
+#endif
+	// uint = 4 little endian bytes
+	// string = (array<char> where the last char must = 0) or (length = 0)
+
+	int32_t error;
+	const ep_char16_t *error_message;
+};
+
+#if !defined(DS_INLINE_GETTER_SETTER) && !defined(DS_IMPL_DUMP_PROTOCOL_GETTER_SETTER)
+struct _DiagnosticsGenerateCoreDumpResponsePayload {
+	uint8_t _internal [sizeof (struct _DiagnosticsGenerateCoreDumpResponsePayload_Internal)];
+};
+#endif
+
```

Comment:
You can define a getter and setter for `error` and `error_message` like so:
```c
DS_DEFINE_GETTER(DiagnosticsGenerateCoreDumpResponsePayload *, generate_core_dump_response_payload, int32_t, error);
DS_DEFINE_SETTER(DiagnosticsGenerateCoreDumpResponsePayload *, generate_core_dump_response_payload, int32_t, error);
DS_DEFINE_GETTER(DiagnosticsGenerateCoreDumpResponsePayload *, generate_core_dump_response_payload, const ep_char16_t *, error_message);
DS_DEFINE_SETTER(DiagnosticsGenerateCoreDumpResponsePayload *, generate_core_dump_response_payload, const ep_char16_t *, error_message);
```
By convention, we typically include `alloc`, `free`, and `init` methods for the type.

Reply:
Do I really need to add setters, alloc/free/init (and size) functions for this? I was really trying to keep this simple since it onlyh has 2 members.

Reply:
You don't _need_ them, but the convention in the EventPipe code is to have them. I'd be fine with you leaving them off if you want. @lateralusX I can't think of a reason he needs to do this short of being consistent with convention.